### ISSUE TITLE
how to use with dynamically added content?

### ISSUE THREAD

**Issue by nickcom111** [PrismJS/prism#2447]
if this pre tag is part of the HTML then everything works as intended - the javascript script file contents is displayed (thanks to the File Highlight plugin) and it is formatted nicely:

`<pre data-src="example_script.js"></pre>;`

However, if I dynamically append this same pre tag to the DOM instead (for example using jQuery) then nothing happens at all - the script is not made visible and nothing is formatted:

```
$(body).append(<pre data-src="stream/samples/inmail.sql"></pre>;
Prism.highlightAll();
```

I'm sure I must be missing something obvious? Appreciate any help

**Comment by RunDevelopment** [comment]
Unfortunately, I can't tell you where the problem lies with this little info.

Could you please share a project that reproduces this behavior? (It doesn't have to be the actual project, just anything that reproduces your issue.)

**Comment by nickcom111** [comment]
sure - here is an example: https://jsfiddle.net/es4bgxtq/5/

thanks for your help here! This tool is amazing!

**Comment by RunDevelopment** [comment]
A few things:

1. Please don't mix versions. We only guarantee Prism components of the same version working with each other.
2. You only need to import one theme. `themes/prism.css` is _not_ a base theme which all others need. It's just the default theme. Importing `themes/prism-coy.css` alone is enough. (`themes/prism.css` might even interfere with the other theme but that doesn't seem to be the case with Coy).

And lastly, the interesting part:
This took me way longer to debug than it should have. The problem is that the in Prism v1.20.0 (and before), File Highlight attaches itself to [`DOMContentLoaded`](https://github.com/PrismJS/prism/blob/c6c62a69a06e80535c387f4041da80e23a3654f0/plugins/file-highlight/prism-file-highlight.js#L84) and since you run your dynamic code after the DOM is loaded, File Highlight doesn't do anything for dynamic content. We finally merged #1974, so this won't be an issue in the next release but right now, you have to call `Prism.fileHighlight()` instead of `Prism.highlightAll` to highlight all `<pre data-src="...">` elements. If you also have regular code elements, you have to call `Prism.highlightAll` and then `Prism.fileHighlight`.

[Here's](https://jsfiddle.net/5bxernpz/) a fixed version of your example.
